src/corehost/cli/test/nativehost/comhost_test.cpp

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "comhost_test.h"
+#include <error_codes.h>
+#include <iostream>
+#include <future>
+#include <pal.h>
+
+namespace
+{
+    using DllGetClassObject_fn = HRESULT(STDMETHODCALLTYPE*)(REFCLSID rclsid, REFIID riid, LPVOID *ppv);
+
+    class comhost_exports
+    {
+    public:
+        comhost_exports(const pal::string_t &comhost_path)
+        {
+            if (!pal::load_library(&comhost_path, &_dll))
+            {
+                std::cout << "Load library of comhost failed" << std::endl;
+                return;
+            }
+
+            get_class_obj_fn = (DllGetClassObject_fn)pal::get_symbol(_dll, "DllGetClassObject");
```

Comment:
Instead of the type, you can use `decltype(get_class_obj_fn)`. Definitely not a requirement but I find it less error prone.